bee-message/src/message.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    parents::Parents,
+    payload::{option_payload_pack, option_payload_packed_len, option_payload_unpack, Payload},
+    Error, MessageId,
+};
+
+use bee_common::packable::{Packable, Read, Write};
+use bee_pow::providers::{miner::Miner, NonceProvider, NonceProviderBuilder};
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+
+/// The minimum number of bytes in a message.
+pub const MESSAGE_LENGTH_MIN: usize = 53;
+
+/// The maximum number of bytes in a message.
+pub const MESSAGE_LENGTH_MAX: usize = 32768;
+
+/// Represent the object that nodes gossip around the network.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Message {
+    /// Specifies which network this message is meant for.
+    network_id: u64,
+    /// The [MessageId]s that this message directly approves.
```

Comment:
I am not sure but I think you also need the backticks around `MessageId` here? maybe it's not needed anymore in current rust.